Automattic/sensei
# Fix private message emails not being sent

Fixes #6188

### Changes proposed in this Pull Request

Move the `sensei_new_private_message` hook to make sure it's firing every time a new private message is created. This fixes the emails but it introduced an unnecessary redirect request (not noticeable because of AJAX) in learning mode. This PR fixes the redirect as well.

### Testing instructions

Testing the email:

* Create a course, enable learning mode, and assign it to a teacher.
* Take the course as a student and send a message to the teacher.
* Make sure the message is received in the backend.
* Make sure an email is sent to the teacher.

Testing no unnecessary redirects in learning mode:

* Create a course, enable learning mode, and assign it to a teacher.
* Take the course as a student and send a message to the teacher.
* There should be no redirects made after the send message AJAX request. To verify this, use the network tab in devtool.

Testing the success notice in classic mode:

* Create a course, disable learning mode, and assign it to a teacher.
* Add the "Contact teacher" block to the course and to a lesson.
* Take the course as a student and send a message to the teacher from the course and from the lesson page.
* You should see a `Your private message has been sent.` notice at the top.
